### DISCUSSION TITLE
cURL error 6: Could not resolve host: ob.nordigen.com

### DISCUSSION THREAD

**Discussion by brown5tick** [firefly-iii/firefly-iii#7946] [Firefly III]
What am I doing wrong here, folks?

This morning I've:
+ Updated FF3 to v6.0.23 from v5.7.13
+ Updated FIDI to v1.3.8 from 0.9.17
+ Renewed my FF3 Personal Access Token
+ Renewed my Nordigen/GoCardless Secret ID & Key
+ Validated connectivity by running the following command on my SynologyNAS (via SSH) and receiving a valid response:
`curl -X POST "https://ob.nordigen.com/api/v2/token/new/" -H "accept: application/json" -H "Content-Type: application/json" -d "{\"secret_id\":\"string\", \"secret_key\":\"string\"}"`

But I still can't use Nordigen in FIDI.

FIDI interface reports: "The Nordigen API is configured incorrectly and cannot be used to import data."

Docker container log reports:

`2023-09-10 13:55:07] production.ERROR: cURL error 6: Could not resolve host: ob.nordigen.com (see https://curl.haxx.se/libcurl/c/libcurl-errors.html) for https://ob.nordigen.com/api/v2/token/new/ {"exception":"[object] (GuzzleHttp\\Exception\\ConnectException(code: 0): cURL error 6: Could not resolve host: ob.nordigen.com (see https://curl.haxx.se/libcurl/c/libcurl-errors.html) for https://ob.nordigen.com/api/v2/token/new/ at /var/www/html/vendor/guzzlehttp/guzzle/src/Handler/CurlFactory.php:210) | stdout`

Other/previous cURL calls for other purposes e.g. jquery, bootstrap etc...

What am I missing!?

**Comment by JC5** [comment]
That's a DNS error. Which DNS service are you using?

**Reply by brown5tick** [reply]
Yeah, DNS issue certainly. I wondered if it was due to the rebranding to GoCardless but they're still supporting the old domain name.

I'm using AdGuard but it's reporting that all is well. I disabled it to check if it was interfering in some way but it didn't make any difference. Here's an example of the AdGuard log output:

<img width="1184" alt="image" src="https://github.com/firefly-iii/firefly-iii/assets/144599640/e12cf569-8cc0-4ef8-abe7-b8a10cc3de1b">

**Reply by JC5** [reply]
Weird. Is Docker also using Adguard? It has its own dns settings in /etc/docker/daemon.json

**Reply by brown5tick** [reply]
The only file at the location you mention is _key.js._ Am I looking in the wrong place?

...but I believe it does use it, yeah. DHCP issues it as the default DNS gateway. What's curious though is that I get the same issue whether AdGuard is enabled or disabled.

**Reply by JC5** [reply]
It's the right place, but if the file doesn't exist it shouldn't be of influence. It's weird that it works with and without. Which DNS service is behind your adguard?

**Reply by brown5tick** [reply]
Umm... I dunno. It's just the DNS service provided my my ISP (Ziggo, NL).

What's baffling me is that independently of FIDI, the cURL command is succeeding. The connectivity is there. The DNS resolution is there.

Are there any other troubleshooting steps I can take to try and narrow this down? Without FIDI, FIII is all but useless.

**Reply by JC5** [reply]
What happens when you use the `dig` command from inside the container itself?

**Reply by brown5tick** [reply]
Wow. I've gone way down the rabbit-hole on this one.

I found out how to access the container's console and how to install the `dig` command via Debian dnstools. The interesting and perhaps unsurprising outcome though was that I got a DNS resolution error when trying to connect to _deb.debian.org,_ suggesting the problem wasn't limited to Nordigen. Made me wonder if or how the other cURLs in the initialisation sequence were working btw.

So I began to think it was a Docker/Container config problem. All of my containers are hanging off the standard Bridge network in Docker. I looked into how that relates to the underlying host network and found that the DNS config is inherited from the host's `/etc/resolv.conf` file... which all looked pukka.

I saw in AdGuard that the deb.debian.org calls were also registering as successful in the query logs. Rather than just simply disabling AdGuard, I routed the host's DNS calls back to the primary gateway instead of the AdGuard server. Lo and behold: problem solved. It's definitely an AdGuard problem... or more specifically, a problem with how AdGuard works with Docker containers.

So, probably no longer relevant to this forum (unless anyone has any helpful pointers 😁)
Thanks for helping me with this @JC5 🤜🤛

**Reply by JC5** [reply]
Hah! That's quite the story! 😅

I'm glad you found the issue. And learned a lot of new things. Smarter ever day 🥳
